arangod/Agency/AgencyComm.cpp

```diff
@@ -177,13 +177,54 @@ AgencyOperation::AgencyOperation(std::string const& key,
   _opType.value = opType;
 }
 
+AgencyOperation::AgencyOperation(std::string const& key, AgencyValueOperationType opType,
+                                 std::shared_ptr<velocypack::Builder> value)
+    : AgencyOperation(std::move(key), opType, value->slice()) {
+  _holder = std::move(value);
+}
+
 AgencyOperation::AgencyOperation(std::string const& key, AgencyValueOperationType opType,
                                  VPackSlice newValue, VPackSlice oldValue)
     : _key(AgencyCommHelper::path(key)), _opType(), _value(newValue), _value2(oldValue) {
   _opType.type = AgencyOperationType::Type::VALUE;
   _opType.value = opType;
 }
 
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path)
+    : _key(path->str()), _opType() {
+  _opType.type = AgencyOperationType::Type::READ;
+}
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path,
+                                 AgencySimpleOperationType opType)
+    : _key(path->str()), _opType() {
+  _opType.type = AgencyOperationType::Type::SIMPLE;
+  _opType.simple = opType;
+}
+
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path,
+                                 AgencyValueOperationType opType,
+                                 velocypack::Slice const value)
+    : _key(path->str()), _opType(), _value(value) {
+  _opType.type = AgencyOperationType::Type::VALUE;
+  _opType.value = opType;
+}
+
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path,
+                                 AgencyValueOperationType opType,
+                                 std::shared_ptr<velocypack::Builder> value)
+    : AgencyOperation(path, opType, value->slice()) {
+  _holder = std::move(value);
```

Comment:
I don't think this is safe, as `_value` will point into the slice held by the old version of `_holder`, which is overriden here.

Reply:
As above, it should be okay with the delegating constructor. I've also added an assertion here now to clarify that this works as intended.

Reply:
Ok, thanks. Too many constructors here...